From 7a9939f714bcafe8d6616bfbdeb87d2e4b1f24e8 Mon Sep 17 00:00:00 2001
From: pitpompej <pitpompej@arcor.de>
Date: Fri, 8 Mar 2013 21:05:31 +0100
Subject: [PATCH] Prevent timeout error because of waiting for port settings
 change event on the wrong pipeline object when using
 deinterlace mode

---
 xbmc/cores/omxplayer/OMXVideo.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/cores/omxplayer/OMXVideo.cpp b/xbmc/cores/omxplayer/OMXVideo.cpp
index d4b8fbf..15bc8fa 100644
--- a/xbmc/cores/omxplayer/OMXVideo.cpp
+++ b/xbmc/cores/omxplayer/OMXVideo.cpp
@@ -885,10 +885,10 @@ int COMXVideo::Decode(uint8_t *pData, int iSize, double dts, double pts)
           {
             CLog::Log(LOGERROR, "%s::%s - error m_omx_image_fx.SetParameter(OMX_IndexParamPortDefinition) omx_err(0x%08x)\n", CLASSNAME, __func__, omx_err);
           }
-          omx_err = m_omx_decoder.WaitForEvent(OMX_EventPortSettingsChanged);
+          omx_err = m_omx_image_fx.WaitForEvent(OMX_EventPortSettingsChanged);
           if(omx_err != OMX_ErrorNone)
           {
-             CLog::Log(LOGERROR, "%s::%s - error m_omx_decoder.WaitForEvent(OMX_EventPortSettingsChanged) omx_err(0x%08x)\n", CLASSNAME, __func__, omx_err);
+             CLog::Log(LOGERROR, "%s::%s - error m_omx_image_fx.WaitForEvent(OMX_EventPortSettingsChanged) omx_err(0x%08x)\n", CLASSNAME, __func__, omx_err);
           }
           port_image.nPortIndex = m_omx_image_fx.GetOutputPort();
           omx_err = m_omx_image_fx.GetParameter(OMX_IndexParamPortDefinition, &port_image);
-- 
1.7.10

